### ISSUE TITLE
"vagrant destroy confirmation" - [yN]

### ISSUE THREAD

**Issue by lloy0076** [hashicorp/vagrant#3027]
# TL;DR

Typing "yes" in response to the question regarding destroying the machine, gives the unexpected (to a human) response of _not_ destroying the machine.

Consider changing:

> Are you sure you want to destroy the _name_ VM [y/N]? 

to 

> Are you sure you want to destroy the _name_ VM [confirm/CANCEL]?

...and accept only "confirm" to proceed.

Yes/No questions need a human to determine precisely what is being asked. Cancel is unambiguous. Alternatively make the response [yes/NO] and require the response 'yes' (not just 'y') and indicate with an appropriate error message if a user just responds 'y'.

At the very least, DO NOT say the confirmation was declined if it was not declined. Say something like, "The VM _name_ was not destroyed, as un unknown response of RESPONSE was received. Valid response are 'y' or 'n'."
# Long Version

When destroying a box, we see:

> isengaard:AGIX-Centos lloy0076$ vagrant destroy
> Are you sure you want to destroy the 'default' VM? [y/N] yes
> [default] The VM 'default' will not be destroyed, since the confirmation
> was declined.

Last time I looked, yes means "yes". It does the same thing with "no", which at least doesn't perform something unexpected.

Some tools will refuse to accept just "y" or "n" (IIRC, emacs is one of them) when they want "yes" or "no".

Possible solutions and what they look like:

> isengaard:AGIX-Centos lloy0076$ vagrant destroy
> Are you sure you want to destroy the 'default' VM? [y/N] yes
> [error] The VM 'default' will not be destroyed, you must respond
> with 'y' or 'n'.

That is possible and, imho, confusing. The "N" is capitalised and should we respond "N" or "n".

> isengaard:AGIX-Centos lloy0076$ vagrant destroy
> Are you sure you want to destroy the 'default' VM? [yes/NO] y
> [default] The VM 'default' will not be destroyed, you must respond
> with 'yes' or 'no' (capitalisation does not matter).

This is better.

However, "yes" and "no" involve the operator to think (or worse, a computer to think).
## Preferred Option

> isengaard:AGIX-Centos lloy0076$ vagrant destroy
> Are you sure you want to destroy the 'default' VM? [confirm/CANCEL] confirm
> [default]: ... the it was destroyed message ...

And:

> isengaard:AGIX-Centos lloy0076$ vagrant destroy
> Are you sure you want to destroy the 'default' VM? [confirm/CANCEL] cancel 
> [default] The VM 'default' was not be destroyed, since the operation was
> cancelled.

Please note the change to past tense.

And:

> isengaard:AGIX-Centos lloy0076$ vagrant destroy
> Are you sure you want to destroy the 'default' VM? [confirm/CANCEL] confrm
> [default] The VM 'default' was not be destroyed, as an unexpected response
> 'confrm' was received. Please respond 'confirm' to destroy the machine;
> the destroy operation will not proceed if any other response is received.

Again, note change to past tense.
